### PR TITLE
feat: refactoring & cleanup

### PR DESCRIPTION
**Pull request by Jaszkowic**
This is my humble try of organizing and documenting the DWD harvester better. Functionally, nothing changed.

- Split code into meaningful functions documented with `docstring` [https://peps.python.org/pep-0257/](docstring)
- Make database queries more readable
- Organize file structure
- Update readme
- Handle environment variables in one file only
- Reference resources at points where external libraries are used
- remove unused files
- Generate the buffered shape file instead of storing it in the repository

### CODE REVIEW

File: README.md
Comment (ff6347) [inline README.md:1]: There are a few new env varaibles. Would be nice to have documentation for them

File: harvester/Dockerfile [hunk 1]
```diff
@@ -27,11 +27,6 @@ COPY --from=builder /install /usr/local
...
 
-# COPY harvester.py /app/
-# COPY prepare.py /app/
-# COPY grid/ grid/
-# COPY assets/ /app/assets
 COPY . /app/
 
-
-CMD python /app/harvester.py && python /app/mapbox_tree_update.py
+CMD cd /app/prepare && SHAPE_RESTORE_SHX=YES python create-buffer.py && cd ../src && python run_harvester.py
```
Comment (ff6347) [inline harvester/Dockerfile:32]: `CMD` should not contain directory switching IMO. Preparation should be a different `RUN` step. CMD and ENTRYPOINT can be replaced/overwritten when using the image in a different way.

File: harvester/prepare/.vscode/launch.json
Comment (ff6347) [inline harvester/prepare/.vscode/launch.json:1]: Do we need these? Editor settings should be a user thing

File: harvester/sample.env
Comment (ff6347) [inline harvester/sample.env:1]: New env vars are not in sample.env

File: harvester/src/supabase_utils.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+            requests.put
+            if check_file_exists_in_supabase_storage(
+                supabaseUrl, supabaseBucketName, file_name
+            )
+            else requests.post
+        )
+        response = http_method(
+            file_url,
+            files={"file": file},
+            headers={
```
Comment (ff6347) [inline harvester/src/supabase_utils.py:30]: Dies this work? Isn't the `apikey` header missing? See https://supabase.com/docs/guides/api/quickstart

```
curl 'https://<PROJECT_REF>.supabase.co/rest/v1/todos' \
-H "apikey: <ANON_KEY>" \
-H "Authorization: Bearer <ANON_KEY>"
```

**Review by ff6347 — CHANGES_REQUESTED** [review]
Sorry for the super late review. I didn't run it locally and I did not read go into the logic yet. These are things noticed on first look. I like the modularization of it all.

File: README.md
Comment (Jaszkowic) [reply README.md:1]: will add

File: harvester/src/supabase_utils.py (hunk 2 above)
Comment (ff6347) [reply harvester/src/supabase_utils.py:30]: But it also needs the `apikey` header

Comment (ff6347) [reply harvester/src/supabase_utils.py:30]: Wired. Actually it should not.

Comment (ff6347) [reply harvester/src/supabase_utils.py:30]: Tested it. Works in local instance but not with remote.

Comment (Jaszkowic) [reply harvester/src/supabase_utils.py:30]: (for future reference) As discussed, this works without api key:

```
curl --location 'https://<supbase_project_id>.supabase.co/storage/v1/object/data_assets/test.txt' \
--header 'Authorization: Bearer service_role_key' \
--form '=@"/path/to/file"'
```

File: harvester/src/mapbox_tree_update.py [hunk 3]
```diff
@@ -81,7 +81,7 @@ def generate_trees_csv(temp_dir, db_conn):
         lines = []
         for tree in tqdm(trees):
             age = int(current_year) - int(tree[4]) if tree[4] != 0 else ""
-            line = "{},{},{},{},{}".format(
+            line = "{},{},{},{},{},{}".format(
```
Comment (ff6347) [inline harvester/src/mapbox_tree_update.py]: Can't we use f strings for things like these? 

    line = f"{tree[0]},{tree[1]},{tree[2]},{tree[3]},{age},{tree[5]}"

**Review by ff6347 — APPROVED** [review]
if it works against staging – merge it

File: harvester/src/mapbox_tree_update.py (hunk 3 above)
Comment (ff6347) [reply harvester/src/mapbox_tree_update.py]: Not that important. But in general the f strings are easier to read.

Comment (Jaszkowic) [reply harvester/src/mapbox_tree_update.py]: changed in latest commit

**Pull request merged**